### PR TITLE
[Android Auto] Fix android automotive surface recreation issue

### PR DESCRIPTION
**Pull request by kmadsen**
### Summary of changes

When building Android Auto for Android Automotive, the automotive surface library has a different behavior for creating surfaces when transitioning screens. 

Tested and reproduced on `androidx.car.app:app-automotive:1.1.0` and `androidx.car.app:app-automotive:1.2.0-rc`.

### User impact (optional)

Fixes a blocking issue for anyone trying to build an Android Auto app for Android Automotive

cc: @yunikkk @pengdev 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: Resolves https://github.com/mapbox/mapbox-maps-android/issues/1503
Related: https://github.com/mapbox/mapbox-maps-android/issues/1397

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: scripts/check-changelog.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ if grep -q "$pr_number" <<< "$skip_changelog_prs"; then
   exit 0
 fi
 
-changelog_diff=$(git diff origin/main..HEAD CHANGELOG.md)
+changelog_diff=$(git diff origin/main..HEAD CHANGELOG.md extension-androidauto/CHANGELOG.md)
```
Comment (pengdev) [inline scripts/check-changelog.sh:27]: nice catch

File: extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt [hunk 2]
```diff
@@ -56,7 +56,14 @@ internal class CarMapSurfaceOwner(
     logI(TAG, "surfaceAvailable")
     val oldCarMapSurface = this.mapboxCarMapSurface
     this.mapboxCarMapSurface = mapboxCarMapSurface
-    oldCarMapSurface?.let { carMapObservers.forEach { it.onDetached(oldCarMapSurface) } }
+    oldCarMapSurface?.let {
+      carMapObservers.forEach { it.onDetached(oldCarMapSurface) }
+      // The surface can become available without a call to surfaceDestroyed
+      // https://issuetracker.google.com/issues/235121269
```
Comment (pengdev) [inline extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt:62]: do we need to revert this change after the upstream issue is fixed? If yes, we should probably open a new issue to track the tailwork.

**Review by pengdev — APPROVED** [review]
LGTM 🚢

File: extension-androidauto/src/test/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwnerTest.kt [hunk 3]
```diff
@@ -224,13 +224,12 @@ class CarMapSurfaceOwnerTest {
       observer.onAttached(firstSurface)
       observer.onVisibleAreaChanged(any(), any())
       observer.onDetached(firstSurface)
+      firstMapSurface.onStop()
+      firstMapSurface.surfaceDestroyed()
+      firstMapSurface.onDestroy()
       observer.onAttached(secondSurface)
       observer.onVisibleAreaChanged(any(), any())
     }
-    // Map style changes should not destroy the map.
```
[the remark this answers is not in the document]
Comment (pengdev) [reply extension-androidauto/src/test/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwnerTest.kt:230]: Does the style change destroy the surface currently?

**Review by yunikkk — APPROVED** [review]

File: extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt [hunk 4]
```diff
@@ -56,7 +56,14 @@ internal class CarMapSurfaceOwner(
...
+      oldCarMapSurface.mapSurface.onStop()
+      oldCarMapSurface.mapSurface.surfaceDestroyed()
+      oldCarMapSurface.mapSurface.onDestroy()
+    }
```
Comment (yunikkk) [inline extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt:66]: Seems it mostly duplicates the logic at `fun surfaceDestroyed()` below, maybe extract common logic to a separate method?

File: extension-androidauto/src/test/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwnerTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (kmadsen) [reply extension-androidauto/src/test/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwnerTest.kt:230]: > Does the style change destroy the surface currently?

No, style changes can be made repeatedly without destroying the surface. The original approach was to notify "onAttached/onDetached" on style change without destroying the surface (this test insured that the surface would not be destroyed).

The load style is happening after the surface is created https://github.com/mapbox/mapbox-maps-android/blob/b75621b4f3cc9178f79750fba8941fe3819f68d3/android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/CarMapShowcase.kt#L25

File: extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt (hunk 2 above)
Comment (kmadsen) [reply extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt:62]: I'm not sure if we will ever revert it, I think we should first wait for a response on the issuetracker.

There have been reports of this happening in production for android auto head units, where the surface is re-created without being destroyed. This may be a safeguard that we keep, considering it is supporting both AA and AAOS

File: extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt (hunk 4 above)
Comment (kmadsen) [reply extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt:66]: I'm not passionate either way, but my habit is the rule of three. this block also has a issue attached to it so it was not too important to share functions.

Also wondered if the function should live in the mapSurface. AA is not the only downstream implementation that will need to call onStop, surfaceDestroyed, and onDestroy?
